src/Nethermind/Nethermind.JsonRpc/Modules/Admin/AdminRpcModule.cs

```diff
@@ -139,19 +144,31 @@ public ResultWrapper<bool> admin_setSolc()
         return ResultWrapper<bool>.Success(true);
     }
 
+    public ResultWrapper<bool> admin_isStateRootAvailable(BlockParameter block)
+    {
+        SearchResult<BlockHeader> headerSearchResult = _blockTree.SearchForHeader(block);
+        if (headerSearchResult.Object is null)
+        {
+            return ResultWrapper<bool>.Fail("Unable to find block. Unable to know state root to verify.");
+        }
+
+        return ResultWrapper<bool>.Success(_stateReader.HasStateForBlock(headerSearchResult.Object));
+    }
+
     public ResultWrapper<PruningStatus> admin_prune()
     {
         return ResultWrapper<PruningStatus>.Success(_pruningTrigger.Trigger());
     }
 
-    public ResultWrapper<string> admin_verifyTrie()
+    public ResultWrapper<string> admin_verifyTrie(BlockParameter block)
     {
-        if (_blockTree.Head is null)
+        SearchResult<BlockHeader> headerSearchResult = _blockTree.SearchForHeader(block);
+        if (headerSearchResult.Object is null)
```

Comment:
I think we need a check if state root is available?

Reply:
done

Reply:
done